neuprint/neuroncriteria.py

```diff
@@ -438,45 +492,34 @@ def __repr__(self):
 
         return s
 
-
     MAX_LITERAL_LENGTH = 3
     assert MAX_LITERAL_LENGTH >= 3, \
         ("The logic in where_expr() assumes valuevars "
          "have length 3 (assuming one could be None).")
 
-
     def global_vars(self):
         exprs = {}
 
-        if len(self.bodyId) > self.MAX_LITERAL_LENGTH:
-            values = [*filter(lambda s: s is not None, self.bodyId)]
-            var = f"{self.matchvar}_search_bodyIds"
-            exprs[var] = (f"{[*values]} as {var}")
-
-        # Never store regular expressions into a global
-        if not self.regex and len(self.type) > self.MAX_LITERAL_LENGTH:
-            values = [*filter(lambda s: s is not None, self.type)]
-            var = f"{self.matchvar}_search_types"
-            exprs[var] = f"{[*values]} as {var}"
-
-        if not self.regex and len(self.instance) > self.MAX_LITERAL_LENGTH:
-            values = [*filter(lambda s: s is not None, self.instance)]
-            var = f"{self.matchvar}_search_instances"
-            exprs[var] = f"{[*values]} as {var}"
-
-        if len(self.status) > self.MAX_LITERAL_LENGTH:
-            values = [*filter(lambda s: s is not None, self.status)]
-            var = f"{self.matchvar}_search_statuses"
-            exprs[var] = f"{[*values]} as {var}"
-
-        if len(self.cellBodyFiber) > self.MAX_LITERAL_LENGTH:
-            values = [*filter(lambda s: s is not None, self.cellBodyFiber)]
-            var = f"{self.matchvar}_search_CBFs"
-            exprs[var] = f"{[*values]} as {var}"
+        for key in self.list_props:
+            values = getattr(self, key)
+            if len(values) > self.MAX_LITERAL_LENGTH:
+                if key.startswith('_'):
+                    key = key[1:]
```

Comment:
This is to handle `class_`, right?

Reply:
Sorry, yes. I originally called it `_class`. Will fix tomorrow!